### PR TITLE
feat(ui5-step-input): implement sap_horizon theme

### PR DESCRIPTION
**Pull request by Todor-ads**
feat(ui5-step-input): implement sap_horizon theme

### CODE REVIEW

**Comment by ilhan007** [comment]
# Check disabled state
openui5 - has border-bottom, more visible
Screenshot 2021-11-04 at 08 05 07

wc - no border-bottom, almost invisible
Screenshot 2021-11-04 at 08 05 24

-> check if you should add border or not and if the opacity/background color is fine

# Readonly (but this is for all themes, not sap_horizon only)
wc - has -/+ icons, no not-editable icon (that has to be present in sap_horizon only)
Screenshot 2021-11-04 at 08 06 57

openui5 - no -/+ icons, has not-editable icon
Screenshot 2021-11-04 at 08 07 10

-> here, at least hide +/- icons when readonly

File: packages/main/src/themes/StepInput.css
Comment (niyap) [inline packages/main/src/themes/StepInput.css:135]: In openUI5 when the increase/decrease button is active, there is a box-shadow and I think that it is the expected behaviour according to the requirement for the other inputs. Check the visualisation in openUI5 and make it the same here, when the button is active.

File: packages/main/src/StepInput.hbs
Comment (ilhan007) [inline packages/main/src/StepInput.hbs:57]: you can use unless readonly, instead of if hasNotReadOnly

File: packages/main/src/themes/StepInput.css
Comment (niyap) [reply packages/main/src/themes/StepInput.css:135]: If you hover over the increase/decrease button, the component box-shadow in not visible. When you hover over the buttons, it should be the same as you hover over the input field.

File: packages/main/src/themes/StepInput.css
Comment (niyap) [inline packages/main/src/themes/StepInput.css:203]: When the component is disabled, it should have opacity added over the regular state. Which means that we should have the same background image as in regular state and opacity on top

**Review by niyap — CHANGES_REQUESTED** [review]

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-b should the UI5 Web Components core team take over and address the comments to complete the implementation of the StepInput. We would like to complete this at least tomorrow 5 Nov, Fri (although it should have been completed until 3, Wed)

File: packages/main/src/StepInput.hbs
Comment (Todor-ads) [reply packages/main/src/StepInput.hbs]: done

File: packages/main/src/themes/StepInput.css
Comment (Todor-ads) [reply packages/main/src/themes/StepInput.css]: done

File: packages/main/src/themes/StepInput.css
Comment (Todor-ads) [reply packages/main/src/themes/StepInput.css:193]: done

File: packages/main/src/themes/StepInput.css
Comment (Todor-ads) [reply packages/main/src/themes/StepInput.css:248]: done

File: packages/main/src/StepInput.js
Comment (niyap) [inline packages/main/src/StepInput.js:190]: Focused

File: packages/main/src/StepInput.js
Comment (niyap) [inline packages/main/src/StepInput.js:545]: It is strange, on the first line you add focused attribute, on the second one you remove it?

File: packages/main/src/StepInput.js
Comment (niyap) [inline packages/main/src/StepInput.js:196]: I could not understand why you add "focused" when you already have a property with the same function called "_focused"

File: packages/main/src/themes/StepInput.css
Comment (niyap) [inline packages/main/src/themes/StepInput.css:70]: Are you sure that the parameter resolves the correct value, since there is a typo

File: packages/main/src/themes/StepInput.css
Comment (niyap) [inline packages/main/src/themes/StepInput.css:176]: underscore for consistency

File: packages/main/src/themes/base/StepInput-parameters.css
Comment (niyap) [inline packages/main/src/themes/base/StepInput-parameters.css:6]: You have to change it here as well

File: packages/main/src/themes/sap_horizon/StepInput-parameters.css
Comment (niyap) [inline packages/main/src/themes/sap_horizon/StepInput-parameters.css:5]: Here, the typos are two - dash between the ui5 and step and then "backgroun"

File: packages/main/src/themes/StepInput.css
Comment (niyap) [inline packages/main/src/themes/StepInput.css:25]: Even, with the fix in the input for the background image when disabled, it does not come out of the box below the buttons, you should adjust it

File: packages/main/src/StepInput.js
Comment (Todor-ads) [reply packages/main/src/StepInput.js:190]: Done

File: packages/main/src/themes/StepInput.css
Comment (Todor-ads) [reply packages/main/src/themes/StepInput.css:70]: done

File: packages/main/src/themes/StepInput.css
Comment (Todor-ads) [reply packages/main/src/themes/StepInput.css:176]: done

File: packages/main/src/themes/base/StepInput-parameters.css
Comment (Todor-ads) [reply packages/main/src/themes/base/StepInput-parameters.css:6]: done

File: packages/main/src/themes/sap_horizon/StepInput-parameters.css
Comment (Todor-ads) [reply packages/main/src/themes/sap_horizon/StepInput-parameters.css:5]: done

File: packages/main/src/themes/StepInput.css
Comment (Todor-ads) [reply packages/main/src/themes/StepInput.css:25]: done

File: packages/main/src/StepInput.js
Comment (Todor-ads) [reply packages/main/src/StepInput.js]: done

File: packages/main/src/StepInput.js
Comment (Todor-ads) [reply packages/main/src/StepInput.js]: done

**Review by niyap — APPROVED** [review]
With the latest changes, I hope that everything works as expected. I have tested sap horizon, as well as fiori 3 and it looks fine.
I will approve the change, but as I am not part of the team, it will be good if somebody else from the team gives his approval that the change is reviewed from me.

File: packages/main/src/StepInput.hbs
Comment (tsanislavgatev) [inline packages/main/src/StepInput.hbs:30]: Remove comment

File: packages/main/src/StepInput.hbs
Comment (tsanislavgatev) [inline packages/main/src/StepInput.hbs:77]: remove comment

File: packages/main/src/themes/sap_horizon/StepInput-parameters.css
Comment (tsanislavgatev) [inline packages/main/src/themes/sap_horizon/StepInput-parameters.css:18]: Remove empty line

File: packages/main/src/themes/StepInput.css
Comment (tsanislavgatev) [inline packages/main/src/themes/StepInput.css:70]: Can you put the value state values in commas.

Comment (tsanislavgatev) [reply packages/main/src/themes/StepInput.css:70]: Example DatePicker.css
value-state="Success"

File: packages/main/src/themes/sap_horizon/StepInput-parameters.css
Comment (Todor-ads) [reply packages/main/src/themes/sap_horizon/StepInput-parameters.css:18]: done

File: packages/main/src/themes/sap_horizon/StepInput-parameters.css
Comment (tsanislavgatev) [inline packages/main/src/themes/sap_horizon/StepInput-parameters.css:14]: Do we have a parameter for this value?

**Comment by tsanislavgatev** [comment]
I see that we now lost the Error state in HC themes:
image
Can we fix it before merging?

**Comment by tsanislavgatev** [comment]
If we exclude my last two comments, the change can be merged. And comments might be addressed in following issue. Depending on the priority, we can proceed and create followup issues.

**Review by tsanislavgatev — APPROVED** [review]

**Comment by ilhan007** [comment]
@Todor-ads 
I suggest follow up on these visual issues (create a an issue or whatever you prefer), in Warning and Error states on high contrast themes
openui5:
Screenshot 2021-11-05 at 18 46 44
Screenshot 2021-11-05 at 18 46 55

webc
Screenshot 2021-11-05 at 18 47 05

**Pull request merged**
